### DISCUSSION TITLE
Remap ctlmgr Ports 80 & 443

### DISCUSSION THREAD

**Discussion by frag1x** [Freetz-NG/freetz-ng#862] [Allgemein]
Hi,
is it technically possible to remap the default fritzbox ctlmgr ports 80 & 443 to something else?
Background: I would like to use a reverse proxy (HAProxy maybe) on freetz to listen on port 80 & 443 so I can access all my network services with only using the standard ports 80 & 443 and subdomains.

I read that you can theoretically change the ports in ar7.cfg but then they are still somehow not free to use.

So my question is if this is possible (maybe not right now but with some future modifications) or it will never be possible so I have to look to run the reverse proxy somewhere else what I don't like to do tbh.

**Comment by fda77** [comment]
Have you tried if iptables works for you unknown device? 
Other way: preload a library for ctlmgr which changes the ports. Freetz does this for multid with libmultid to make dns/53 usable

**Comment by fraternl** [comment]
I assume he wants to do this on the WAN-port.
Wouldn't it work with port translation with the Freetz port forwarding?
He could have the HAproxy working on 8443 and have a port translation 443:8443 in Freetz Portforwards.

I had such a config running with _pound_ on a DD-WRT router 15 years ago.

He could run split-DNS locally bypassing a proxy alltogether when doing it from the LAN.
I don't know if Freetz-NG supports the overriding of hostnames.
I would be interested in that as well as new versions of 3CX demands it.

**Comment by frag1x** [comment]
Thanks for your replies.
I guess preloading a library would be the best solution, but I think this will be too difficult for me.

No, I only need this in my internal LAN.
For example, at the moment, I acces my freetz via https://freetz.local.example.com:4433, my homer bashboard also running on freetz via https://homer.local.example.com:8443, and the default FritzBox webinterface via  https://fritzbox.local.example.com (default 443). ("freetz" and "homer" are just dns-aliases pointing to the fritzbox/freetz)

I would like to run a reverse proxy listening on port 443 on freetz and redirecting to the desired url+port, so I can access the freetz webif, homer dashboard, the default FritzBox webif and maybe other webservices all without adding the nonstandard port at the end of the url. But everything just in my local network.
But as we know, Port 443 is being used by ctlmgr.

**Comment by fda77** [comment]
You could try to add a 2nd ip to lan (or some other) network interface
` ifconfig lan:0 192.168.178.123 netmask 255.255.255.0 up `
If it works it is possible that avm resets this sometimes (PA via incomming phone call, or dsl reconnect)

I'm using the file `/tmp/flash/mod/multid.start` to set persistent default routes
